File: src/api/tauri/updates.ts
/**
 * Typed IPC for Patch My PC–class software updates.
 */

import { invoke } from '@tauri-apps/api/core';
import type {
  SoftwareUpdate,
  UpdateApplyResult,
} from '../../types/device.types';

export const scanSoftwareUpdates = (): Promise<SoftwareUpdate[]> =>
  invoke<SoftwareUpdate[]>('scan_software_updates');

export const listSoftwareUpdates = (): Promise<SoftwareUpdate[]> =>
  invoke<SoftwareUpdate[]>('list_software_updates');

export const applySoftwareUpdates = (
  updateIds: string[],
  confirm: boolean,
): Promise<UpdateApplyResult> =>
  invoke<UpdateApplyResult>('apply_software_updates', {
    updateIds,
    confirm,
  });

File: src-tauri/src/commands/updates.rs
//! IPC for Patch My PC–class software updates.

use tauri::State;

use crate::error::CoreError;
use crate::models::{SoftwareUpdate, UpdateApplyResult};
use crate::updates;
use crate::AppState;

#[tauri::command]
pub fn scan_software_updates(state: State<'_, AppState>) -> Result<Vec<SoftwareUpdate>, CoreError> {
    let conn = state.conn()?;
    updates::scan_updates(&conn)
}

#[tauri::command]
pub fn list_software_updates(state: State<'_, AppState>) -> Result<Vec<SoftwareUpdate>, CoreError> {
    let conn = state.conn()?;
    updates::list_updates(&conn)
}

#[tauri::command]
pub fn apply_software_updates(
    state: State<'_, AppState>,
    update_ids: Vec<String>,
    confirm: bool,
) -> Result<UpdateApplyResult, CoreError> {
    let conn = state.conn()?;
    updates::apply_updates(&conn, update_ids, confirm)
}
